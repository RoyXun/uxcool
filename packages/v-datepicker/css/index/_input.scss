@mixin input() {
  position: relative;
  display: inline-block;
  width: 100%;
  height: 25px;
  margin: 0 0;
  padding: 4px 10px;
  border-radius: 6px 6px;
  border: 1px solid #d9d9d9;
  background-color: #ffffff;
  color: #666;
  line-height: 1.5;
  transform: border 0.3s cubic-bezier(0.35, 0, 0.25, 1),
    background 0.3s cubic-bezier(0.35, 0, 0.25, 1), box-shadow 0.3s cubic-bezier(0.35, 0, 0.25, 1);
  &:disabled {
    background: #eee;
  }
}
