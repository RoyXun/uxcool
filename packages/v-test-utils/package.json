{
  "name": "@suning/v-test-utils",
  "version": "1.0.10",
  "main": "src/index.js",
  "module": "src/index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run server",
    "compile": "v-tools run compile",
    "server": "v-tools run server"
  },
  "peerDependencies": {
    "vue": ">= 2.5.*"
  },
  "devDependencies": {
    "@suning/v-tools": "^1.0.35",
    "@vue/test-utils": "^1.0.0-beta.16"
  },
  "dependencies": {
    "@suning/v-trigger": "^1.0.64",
    "@vue/server-test-utils": "^1.0.0-beta.30",
    "jquery": "^3.3.1",
    "vue-server-renderer": "^2.6.11"
  }
}
