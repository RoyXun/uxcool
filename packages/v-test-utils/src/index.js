import $ from 'jquery';

export * from '@vue/test-utils';

export { $, $ as jQuery };
// ms
export async function waitTime(delay) {
  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, delay);
  });
}
