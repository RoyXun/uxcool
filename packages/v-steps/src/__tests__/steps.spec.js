import { mount, waitTime, destroyWrapper } from '@suning/v-test-utils';
import { VSteps, VStep } from '../index';

const stepsWrapper = {
  props: {
    current: {
      type: [Number, String],
      default: 1
    },
    direction: {
      type: String,
      default: 'horizontal'
    },
    status: {
      type: String,
      default: 'process'
    },
  },
  render() {
    const { current, direction, status } = this;
    return (
      <VSteps current={current} direction={direction} status={status}>
        <VStep title="已完成" desc="这是描述1"></VStep>
        <VStep title="进行中" desc="这是描述2"></VStep>
        <VStep title="待运行" desc="这是描述3"></VStep>
        <VStep title="待运行" desc="这是描述4"></VStep>
      </VSteps>
    );
  }
};

describe('steps', () => {
  it('render VSteps correctly', async () => {
    Object.defineProperty(HTMLDivElement.prototype, 'offsetWidth', {
      set: () => 400
    });
    Object.defineProperty(window.document, 'documentElement', {
      set: () => {}
    });
    const wrapper = mount(stepsWrapper);
    const steps = wrapper.findAll(VStep);

    await waitTime();
    expect(wrapper.is('.v-steps')).toBe(true);
    expect(steps.length).toBe(4);
    expect(steps.at(0).is('.v-steps-item-finish')).toBe(true);
    expect(steps.at(0).find('.v-steps-item-title').text()).toBe('已完成');
    expect(steps.at(0).find('.v-steps-item-description').text()).toBe('这是描述1');

    wrapper.setProps({
      current: 2,
      status: 'error'
    });
    await waitTime();
    expect(wrapper.html()).toMatchSnapshot();
    expect(steps.at(1).is('.v-steps-item-finish')).toBe(true);
    destroyWrapper(wrapper);
  });
});
