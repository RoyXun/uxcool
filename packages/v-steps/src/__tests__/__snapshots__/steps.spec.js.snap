// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`steps render VSteps correctly 1`] = `
<div class="v-steps v-steps-horizontal v-steps-label-horizontal">
  <div class="v-steps-item v-steps-item-finish">
    <div class="v-steps-item-tail"></div>
    <div class="v-steps-item-icon"><span class="v-steps-icon vicon vicon-ok"><!----></span></div>
    <div class="v-steps-item-content">
      <div class="v-steps-item-title">
        已完成
      </div>
      <div class="v-steps-item-description">
        这是描述1
      </div>
    </div>
  </div>
  <div class="v-steps-item v-steps-item-finish v-steps-next-error">
    <div class="v-steps-item-tail"></div>
    <div class="v-steps-item-icon"><span class="v-steps-icon vicon vicon-ok"><!----></span></div>
    <div class="v-steps-item-content">
      <div class="v-steps-item-title">
        进行中
      </div>
      <div class="v-steps-item-description">
        这是描述2
      </div>
    </div>
  </div>
  <div class="v-steps-item v-steps-item-error">
    <div class="v-steps-item-tail"></div>
    <div class="v-steps-item-icon"><span class="v-steps-icon vicon vicon-close"><!----></span></div>
    <div class="v-steps-item-content">
      <div class="v-steps-item-title">
        待运行
      </div>
      <div class="v-steps-item-description">
        这是描述3
      </div>
    </div>
  </div>
  <div class="v-steps-item v-steps-item-wait">
    <div class="v-steps-item-tail"></div>
    <div class="v-steps-item-icon"><span class="v-steps-icon"><span>
          4
        </span></span></div>
    <div class="v-steps-item-content">
      <div class="v-steps-item-title">
        待运行
      </div>
      <div class="v-steps-item-description">
        这是描述4
      </div>
    </div>
  </div>
</div>
`;
